plotting/plot_handler.py

```diff
@@ -0,0 +1,107 @@
+This class is (will be) responsible for:
+Searching for and retrieving an existing plotting file via the SFTP client
+Getting the associated plotting meta data file
+Instructing the Plotting factory to build an IFrame based on the above
+"""
+import logging
+from utils.clients.sftp_client import SFTPClient
+
+LOGGER = logging.getLogger('app')
+
+
+# pylint:disable=line-too-long, no-else-return
+class PlotHandler:
+    """
+    Utility that takes parameters for a run and (for now) checks if an associated image exists and retrieves it.
+    :param instrument_name: The name of the beamline/spectrometer/instrument.
+    :param rb_number: The ISIS RB number.
+    :param run_number: The run number on the given instrument for the given RB number.
+    :param plot_type:
+        The type of plot file / file extension to be searched for (e.g. "png"). Can be a list of extensions.
+        If None, a list of common graphics file extensions is searched for.
+    """
+
+    def __init__(self, instrument_name, rb_number, run_number, plot_type=None):
+        self.instrument_name = instrument_name
+        self.rb_number = rb_number
+        self.run_number = run_number
+        if plot_type is None:
+            # list of plot types the handler looks for if none is provided; it searches for the types in the order of
+            # the list and selects the first one that exists. In future: could add vector graphics (.pdf, .eps)
+            self.plot_type = ["png", "jpg", "bmp", "gif", "tiff"]
+        elif isinstance(plot_type, str):
+            # if only a single str is passed for a plot type then pass it into a list (so that a single for-loop can be
+            # used to loop over plot types).
+            self.plot_type = [plot_type]
+        else:
+            self.plot_type = plot_type
+        # dictionary of regular expressions for each instrument to describe the possible prefixes in the file name(s)
+        self._instrument_dict = {"MARI": "MAR[I]", "WISH": "WISH"}
+        # path to directory with files for given instrument and RB number
+        self._rbfolder = "/instrument/" + self.instrument_name + "/RBNumber/RB" + str(
+            self.rb_number) + "/" + str(self.run_number) + "/autoreduced/"
+        # parameter to store the file names of any existing plot files found
+        self._existing_plot_files = []
+
+    def _regexp_for_file_name(self, plot_type):
+        """
+        Regular expression used for looking for plot files. It assumes that the file names follow the convention
+        <instrument_abbreviation><run_number>*<.png or other extension>
+        """
+        try:
+            _regexp = self._instrument_dict[self.instrument_name] + str(
+                self.run_number) + "*." + plot_type
+            return _regexp
+        except KeyError:  # if the instrument name does not appear in the dictionary of known instruments
+            LOGGER.info("The instrument name is not recognised")
```

Comment:
Yes just return here - we don't want to raise as it will be live and user exposed on the webapp

Reply:
The pylint 'no-else-return' is actually about the construction:
`if A:
    return A
else:
   do something else`
pylint wants you to write
`if A:
    return A
do something else`
which I personally find less readable.
I have added a return statement to the exception so that a construction of the form 
`if PlotHandler.get_plot_file:`
can be used

Reply:
Yep this is fine